Sources/PostgresNIO/New/PostgresChannelHandler.swift

```diff
@@ -664,6 +727,49 @@ final class PostgresChannelHandler: ChannelDuplexHandler {
         }
     }
 
+    private func prepareStatementComplete(
+        name: String,
+        rowDescription: RowDescription?,
+        context: ChannelHandlerContext
+    ) {
+        let action = self.preparedStatementState.preparationComplete(
+            name: name,
+            rowDescription: rowDescription
+        )
+        guard case .executePendingStatements(let statements, let rowDescription) = action else {
+            preconditionFailure("Expected to have pending statements to execute")
+        }
```

Comment:
we should reconsider the return type from the statemachine here :). No enum needed.